### PR TITLE
Allow DDPM scheduler to use model's predicated variance

### PR DESCRIPTION
**Pull request by eyalmazuz**
This is an implementation of the [Improved Denoising Diffusion Probabilistic Models](https://arxiv.org/pdf/2102.09672.pdf)
which added the ability to learn the variance instead of just predicting the noise.

It is assumed that the step method receives a tensor shape (B, C, ...) when predicting noise.
At the moment, it only implements variance. In order to use it fully, the user needs to write the additional loss function from the paper into their training loop.

For future contribution, I think it would be beneficial to implement a losses module with different loss functions like the VB loss in the Improved Diffusion paper or other loss functions used in different diffusion papers (and maybe a trainer object). If needed, I can make new PRs with the relevant additions, or this can be an ongoing PR (the former seems better in case I'll stop contributing for some reason).

### CODE REVIEW

File: src/diffusers/schedulers/scheduling_ddpm.py [hunk 1]
```diff
@@ -125,6 +134,12 @@ def step(
         generator=None,
     ):
         t = timestep
+        
+        if model_output.shape[1] == sample.shape[1] * 2 and self.variance_type in ["leanred", "learned_range"]:
```
Comment (anton-l) [inline src/diffusers/schedulers/scheduling_ddpm.py:138]:
```suggestion
        if model_output.shape[1] == sample.shape[1] * 2 and self.variance_type in ["learned", "learned_range"]:
```

**Review by anton-l — APPROVED** [review]
Hi @eyalmazuz! Thank you for your contribution, this will be quite useful for when we bring OpenAI's guided diffusion and Glide models into the library! 
We'll have to think a bit more about the API design for learned variance (cc @patil-suraj), but for now this will work fine :)
